### PR TITLE
RWA arraylias conversion

### PR DESCRIPTION
**Pull request by DanPuzzuoli**
### Summary

Converting `rotating_wave_approximation.py` to use arraylias.

I ended up adding the helper functions `_to_dense` and `_to_dense_list` in `arraylias/alias.py` as the `to_array` method had been being used in to convert everything to dense, and we need something analogous. 

### Details and comments

tests that need to be run before merge:

test.dynamics.models (whole folder)
test.dynamics.arraylias (whole folder)
test.dynamics.signals (whole folder)
test.dynamics.pulse (whole folder)

### CODE REVIEW

**Review by to24toro — CHANGES_REQUESTED** [review]
Thanks. Almost looking good to me.

- Regarding alias.py, it might be better to add tests about `_to_dense`and `_to_dense_list`.

File: qiskit_dynamics/models/rotating_wave_approximation.py [hunk 1]
```diff
@@ -235,10 +244,10 @@ def jax_transformable_func(t):
...
 
 def get_rwa_operators(
-    current_ops: Array,
+    current_ops: ArrayLike,
     current_sigs: SignalList,
     rotating_frame: RotatingFrame,
-    frame_freqs: Array,
+    frame_freqs: ArrayLike,
     cutoff_freq: float,
 ):
```
[the remark this answers is not in the document]
Comment (DanPuzzuoli) [reply qiskit_dynamics/models/rotating_wave_approximation.py:252]: The return type for `get_rwa_operators` should be `ArrayLike` and the return type for `get_rwa_signals` is `SignalList`. Just added these type hints.

**Comment by DanPuzzuoli** [comment]
@to24toro I've updated the type hints and added tests for `_to_dense` and `_to_dense_list`.

I had to force push as I'm working from my personal laptop and had forgotten to update the `git config` with my username on commit [7d21240](https://github.com/Qiskit-Extensions/qiskit-dynamics/pull/296/commits/7d21240b02152c9e94cde1868678e8141ca0c3ca), which triggered the CLA message.

File: qiskit_dynamics/models/rotating_wave_approximation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (to24toro) [reply qiskit_dynamics/models/rotating_wave_approximation.py]: Thank you. Regarding `frame_freqs`, it's my mistake and I referred a wrong part.

**Review by to24toro — APPROVED** [review]
LGTM!

**Pull request merged**